src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -256,23 +248,21 @@ void java_bytecode_convert_classt::convert(
 /// java::array[X] where X is byte, float, int, char...
 void java_bytecode_convert_classt::add_array_types(symbol_tablet &symbol_table)
 {
-  // this method only adds stuff to the symbol table, no need to execute it more
-  // than once
-  if(add_array_types_executed)
-    return;
-  add_array_types_executed=true;
-
   const std::string letters="ijsbcfdza";
 
   for(const char l : letters)
   {
     symbol_typet symbol_type=
       to_symbol_type(java_array_type(l).subtype());
 
+    const irep_idt &symbol_type_identifier(symbol_type.get_identifier());
```

Comment:
I don't know, but constructing a reference makes me nervous, I think:

```C++
const irep_idt &symbol_type_identifier=symbol_type.get_identifier();
```

Looks more natural, but this may be nonsense :)